connector/src/main/java/org/jboss/as/connector/services/resourceadapters/IronJacamarActivationResourceService.java

```diff
@@ -0,0 +1,302 @@
+
+
+    private final ManagementResourceRegistration registration;
+    private final Resource deploymentResource;
+    private final boolean statsEnabled;
+
+    protected final InjectedValue<ResourceAdapterDeployment> deployment = new InjectedValue<>();
+
+    protected final InjectedValue<AS7MetadataRepository> mdr = new InjectedValue<AS7MetadataRepository>();
+
+    /**
+     * create an instance *
+     */
+    public IronJacamarActivationResourceService(final ManagementResourceRegistration registration, final Resource deploymentResource,
+                                                final boolean statsEnabled) {
+        super();
+        this.registration = registration;
+        this.deploymentResource = deploymentResource;
+        this.statsEnabled = statsEnabled;
+    }
+
+
+    @Override
+    public void start(StartContext context) throws StartException {
+        ManagementResourceRegistration subRegistration;
+        final CommonDeployment deploymentMD = deployment.getValue().getDeployment();
+        ROOT_LOGGER.infof("Starting IronJacamarActivationResourceService %s", deploymentMD.getDeploymentName());
+
+        try {
+            ResourceBuilder resourceBuilder = ResourceBuilder.Factory.create(SUBSYSTEM_PATH_ELEMENT,
+                    new StandardResourceDescriptionResolver(Constants.STATISTICS_NAME, CommonAttributes.RESOURCE_NAME, CommonAttributes.class.getClassLoader()));
+            subRegistration = registration.registerSubModel(resourceBuilder.build());
+
+        } catch (IllegalArgumentException iae) {
+            subRegistration = registration.getSubModel(PathAddress.pathAddress(SUBSYSTEM_PATH_ELEMENT));
+        }
+        ManagementResourceRegistration ijRegistration;
+
+        try {
+            ijRegistration = subRegistration.registerSubModel(new IronJacamarResourceDefinition());
+
+        } catch (IllegalArgumentException iae) {
+            ijRegistration = subRegistration.getSubModel(PathAddress.pathAddress(Constants.IRONJACAMAR_NAME, Constants.IRONJACAMAR_NAME));
+
+        }
+
+        try {
+            if (deploymentResource != null) {
+
+
+                PathElement peLocalStats = PathElement.pathElement(Constants.STATISTICS_NAME, "extended");
+
+                if (deploymentMD.getConnector() != null && deploymentMD.getConnector().getResourceAdapter() != null) {
+                    ManagementResourceRegistration raRegistration = ijRegistration.
+                            getSubModel(PathAddress.EMPTY_ADDRESS.append(PathElement.pathElement(Constants.RESOURCEADAPTER_NAME)))
+                            .registerOverrideModel(deploymentMD.getDeploymentName(), new OverrideDescriptionProvider() {
+                                @Override
+                                public Map<String, ModelNode> getAttributeOverrideDescriptions(Locale locale) {
+                                    return Collections.emptyMap();
```

Comment:
yup, descriptions are added runtime later registering submodels. I haven't change how resources was registered in previous implementation, moving only them to new location and using PASSIVE service